pallets/phala/src/stakepool.rs

```diff
@@ -35,27 +33,41 @@ pub mod pallet {
 			UnixTime, WithdrawReasons,
 		},
 	};
-	use frame_system::pallet_prelude::*;
+	use frame_system::{pallet_prelude::*, Origin};
+
 	use scale_info::TypeInfo;
 	use sp_runtime::{
-		traits::{Saturating, TrailingZeroInput, Zero},
+		traits::{CheckedSub, Saturating, TrailingZeroInput, Zero},
 		Permill, SaturatedConversion,
 	};
 	use sp_std::{collections::vec_deque::VecDeque, fmt::Display, prelude::*, vec};
 
 	use phala_types::{messaging::SettleInfo, WorkerPublicKey};
 
+	pub use rmrk_traits::primitives::{CollectionId, NftId};
+
 	const STAKING_ID: LockIdentifier = *b"phala/sp";
 
 	const MAX_WHITELIST_LEN: u32 = 100;
 
+	const NFT_PROPERTY_KEY: &str = "stake-info";
+
 	pub struct DescMaxLen;
 
 	impl Get<u32> for DescMaxLen {
 		fn get() -> u32 {
 			400
 		}
 	}
+
+	pub struct ContributeListMaxLen;
```

Comment:
Where is it used?

Reply:
it already exist for a long time，used for whitelist，don‘t know why it shows as new added